packages/web3-rpc-providers/src/web3_provider_publicnode.ts

```diff
@@ -0,0 +1,112 @@
+		[Network.MANTLE_MAINNET]: 'mantle-rpc.publicnode.com',
+		[Network.TAIKO_MAINNET]: 'taiko-rpc.publicnode.com',
+		[Network.TAIKO_HEKLA]: 'taiko-hekla-rpc.publicnode.com',
+		[Network.LINEA_MAINNET]: 'linea-rpc.publicnode.com',
+		[Network.LINEA_SEPOLIA]: 'linea-sepolia-rpc.publicnode.com',
+		[Network.SCROLL_MAINNET]: 'scroll-rpc.publicnode.com',
+		[Network.SCROLL_SEPOLIA]: 'scroll-sepolia-rpc.publicnode.com',
+		[Network.SYSCOIN_MAINNET]: 'syscoin-evm-rpc.publicnode.com',
+		[Network.SYSCOIN_TANENBAUM]: 'syscoin-tanenbaum-evm-rpc.publicnode.com',
+		[Network.HAQQ_MAINNET]: 'haqq-evm-rpc.publicnode.com',
+		[Network.EVMOS_MAINNET]: 'evmos-evm-rpc.publicnode.com',
+		[Network.EVMOS_TESTNET]: 'evmos-testnet-evm-rpc.publicnode.com',
+		[Network.BERACHAIN_TESTNET]: 'berachain-testnet-evm-rpc.publicnode.com',
+		[Network.ETH_MAINNET]: 'ethereum-rpc.publicnode.com',
+		[Network.ETH_SEPOLIA]: 'ethereum-sepolia-rpc.publicnode.com',
+		[Network.ETH_HOLESKY]: 'ethereum-holesky-rpc.publicnode.com',
+		[Network.BSC_MAINNET]: 'bsc-rpc.publicnode.com',
+		[Network.BSC_TESTNET]: 'bsc-testnet-rpc.publicnode.com',
+		[Network.POLYGON_MAINNET]: 'polygon-bor-rpc.publicnode.com',
+		[Network.BASE_MAINNET]: 'base-rpc.publicnode.com',
+		[Network.BASE_SEPOLIA]: 'base-sepolia-rpc.publicnode.com',
+		[Network.ARBITRUM_ONE]: 'arbitrum-one-rpc.publicnode.com',
+		[Network.ARBITRUM_NOVA]: 'arbitrum-nova-rpc.publicnode.com',
+		[Network.ARBITRUM_SEPOLIA]: 'arbitrum-sepolia-rpc.publicnode.com',
+		[Network.AVALANCHE_C_MAINNET]: 'avalanche-c-chain-rpc.publicnode.com',
+		[Network.AVALANCHE_P_MAINNET]: 'avalanche-p-chain-rpc.publicnode.com',
+		[Network.AVALANCHE_X_MAINNET]: 'avalanche-x-chain-rpc.publicnode.com',
+		[Network.AVALANCHE_FUJI_C]: 'avalanche-fuji-c-chain-rpc.publicnode.com',
+		[Network.AVALANCHE_FUJI_P]: 'avalanche-fuji-p-chain-rpc.publicnode.com',
+		[Network.AVALANCHE_FUJI_X]: 'avalanche-fuji-x-chain-rpc.publicnode.com',
+		[Network.OPTIMISM_MAINNET]: 'optimism-rpc.publicnode.com',
+		[Network.OPTIMISM_SEPOLIA]: 'optimism-sepolia-rpc.publicnode.com',
+		[Network.FANTOM_MAINNET]: 'fantom-rpc.publicnode.com',
+		[Network.FANTOM_TESTNET]: 'fantom-testnet-rpc.publicnode.com',
+		[Network.OPBNB_MAINNET]: 'opbnb-rpc.publicnode.com',
+		[Network.OPBNB_TESTNET]: 'opbnb-testnet-rpc.publicnode.com',
+		[Network.GNOSIS_CHIADO]: 'gnosis-chiado-rpc.publicnode.com',
+		[Network.CHILIZ_MAINNET]: 'chiliz-rpc.publicnode.com',
+		[Network.CHILIZ_SPICY]: 'chiliz-spicy-rpc.publicnode.com',
+		[Network.MOONBEAM_MAINNET]: 'moonbeam-rpc.publicnode.com',
+		[Network.BAHAMUT_MAINNET]: 'bahamut-rpc.publicnode.com',
+		[Network.TRON_MAINNET]: 'tron-evm-rpc.publicnode.com',
+		[Network.MOONRIVER_MAINNET]: 'moonriver-rpc.publicnode.com',
+	};
+	// eslint-disable-next-line class-methods-use-this
+	public getRPCURL(network: Network, transport: Transport, _: string, _host: string) {
+		const defaultHost = PublicNodeProvider.networkHostMap[network];
```

Comment:
and concat that here `PublicNodeProvider.networkHostMap[network] + ".publicnode.com"`